compiler/rustc_middle/src/mir/terminator.rs

```diff
@@ -187,7 +248,7 @@ pub enum TerminatorKind<'tcx> {
         /// This allows the memory occupied by "by-value" arguments to be
         /// reused across function calls without duplicating the contents.
         args: Vec<Operand<'tcx>>,
-        /// Destination for the return value. If some, the call is converging.
+        /// Destination for the return value. If none, the call necessarily diverges.
         destination: Option<(Place<'tcx>, BasicBlock)>,
```

Comment:
Must not alias the place given to any Operand::Move passed as argument. All codegen backends depend on this property to allow passing Operand::Move as pointer in case of a type not fitting inside registers without having to male a copy on the stack.

Reply:
I've added in a mention of that in the "needs clarification" section above. Based on the status of the linked issue, it seems that it is still unclear how we are justifying that optimization